lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -313,7 +307,12 @@ def get_data_3d(self):
     @artist.allow_rasterization
     def draw(self, renderer):
         if self._axlim_clip:
-            xs3d, ys3d, zs3d = _viewlim_mask(*self._verts3d, self.axes)
+            mask = np.broadcast_to(
+                _viewlim_mask(*self._verts3d, self.axes),
+                (len(self._verts3d), *self._verts3d[0].shape)
+            )
+            xs3d, ys3d, zs3d = np.ma.array(self._verts3d,
+                                           dtype=float, mask=mask).filled(np.nan)
```

Comment:
Is there a reason to here use this pattern but in l.186 `pos3d[mask] = np.nan`. I believe they do the same thing?

Reply:
The benefit of this pattern is that if `._verts3d`, etc are already masked arrays, then this operation will combine both masks. If we casted to a numpy array and fill with nans, we'd have to first check if it's masked and change the fill value to NaN. Updated the pattern in line 186 to match what's here.